migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,207 @@
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WorkspaceQueryMigration implements DataMigratable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WorkspaceQueryMigration.class);
+
+  private final CatalogFramework catalogFramework;
+
+  private final FilterBuilder filterBuilder;
+
+  private final InputTransformer xmlInputTransformer;
+
+  private static final Security SECURITY = Security.getInstance();
+
+  public WorkspaceQueryMigration(
+      CatalogFramework catalogFramework,
+      FilterBuilder filterBuilder,
+      InputTransformer xmlInputTransformer) {
+    this.catalogFramework = catalogFramework;
+    this.filterBuilder = filterBuilder;
+    this.xmlInputTransformer = xmlInputTransformer;
+  }
+
+  @Override
+  public void migrate() {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+
+    if (systemSubject == null) {
+      LOGGER.debug("An error occurred while attempting to run this migration as admin");
+      return;
+    }
+
+    systemSubject.execute(this::migrateWorkspaces);
+  }
+
+  @VisibleForTesting
+  void migrateWorkspaces() {
+    LOGGER.trace("Beginning workspace query data migration");
+
+    Filter workspaceFilter =
+        filterBuilder.allOf(
+            filterBuilder.attribute(Core.METACARD_TAGS).like().text(WORKSPACE_TAG),
+            filterBuilder.attribute(WORKSPACE_QUERIES).is().like().text("*"));
+
+    Query query = new QueryImpl(workspaceFilter);
+    QueryRequest queryRequest = new QueryRequestImpl(query);
+
+    ResultIterable.resultIterable(catalogFramework::query, queryRequest)
+        .stream()
+        .map(Result::getMetacard)
+        .filter(Objects::nonNull)
+        .forEach(this::migrateWorkspaceMetacard);
+
+    LOGGER.trace("Completed workspace query data migration");
+  }
+
+  private void migrateWorkspaceMetacard(Metacard workspaceMetacard) {
+    LOGGER.trace("Migrating workspace metacard with id [{}]", workspaceMetacard.getId());
+
+    final Attribute queriesAttribute = workspaceMetacard.getAttribute(WORKSPACE_QUERIES);
+    final List<Serializable> queriesAsXml = queriesAttribute.getValues();
+
+    LOGGER.trace(
+        "Beginning migration of {} query metacards from workspace [{}]",
+        queriesAsXml.size(),
+        workspaceMetacard.getId());
+
+    List<Metacard> queryMetacards =
+        queriesAsXml
+            .stream()
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .map(xml -> xml.getBytes(StandardCharsets.UTF_8))
+            .map(ByteArrayInputStream::new)
+            .map(this::xmlToMetacard)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+    LOGGER.trace("Creating query metacards for workspace [{}]", workspaceMetacard.getId());
+
+    CreateResponse response = createMetacards(queryMetacards);
+
+    if (response != null) {
```

Comment:
✏️ Reverse this predicate, i.e. `if (response == null) LOGGER.debug(...); return;` and you can collapse the curly braces.